src/main/java/io/github/thebusybiscuit/slimefun4/core/commands/subcommands/TimingsCommand.java

```diff
@@ -15,27 +21,76 @@
 
 class TimingsCommand extends SubCommand {
 
+    private static final String FLAG_PREFIX = "--";
+    private final Set<String> flags = new HashSet<>(Arrays.asList("verbose"));
+
     TimingsCommand(SlimefunPlugin plugin, SlimefunCommand cmd) {
         super(plugin, cmd, "timings", false);
     }
 
+    @Override
+    protected String getDescription() {
+        return "commands.timings.description";
+    }
+
     @Override
     public void onExecute(CommandSender sender, String[] args) {
         if (sender.hasPermission("slimefun.command.timings") || sender instanceof ConsoleCommandSender) {
-            sender.sendMessage("Please wait a second... The results are coming in!");
-            PerformanceInspector inspector = inspectorOf(sender);
+            if (hasInvalidFlags(sender, args)) {
+                return;
+            }
+
+            boolean verbose = hasFlag(args, "verbose");
+
+            if (verbose && sender instanceof Player) {
+                SlimefunPlugin.getLocalization().sendMessage(sender, "commands.timings.verbose-player", true);
+                return;
+            }
+
+            SlimefunPlugin.getLocalization().sendMessage(sender, "commands.timings.please-wait", true);
+
+            PerformanceInspector inspector = inspectorOf(sender, verbose);
             SlimefunPlugin.getProfiler().requestSummary(inspector);
         } else {
             SlimefunPlugin.getLocalization().sendMessage(sender, "messages.no-permission", true);
         }
     }
 
+    @ParametersAreNonnullByDefault
+    private boolean hasInvalidFlags(CommandSender sender, String[] args) {
+        boolean hasInvalidFlags = false;
+
+        // We start at 1 because args[0] will be "timings".
+        for (int i = 1; i < args.length; i++) {
+            String argument = args[i].toLowerCase(Locale.ROOT);
+
+            if (!argument.startsWith(FLAG_PREFIX) || !flags.contains(argument.substring(2))) {
```

Comment:
This should be startsWith not doesnt. We shouldn't be firing "invalid flag" for a letter a.